x-pack/plugins/cases/public/components/connectors/swimlane/validator.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SwimlaneConnectorType } from '../../../../common';
+import { ValidationConfig } from '../../../common/shared_imports';
+import { CaseActionConnector } from '../../types';
+
+const casesRequiredFields = [
+  'caseIdConfig',
+  'caseNameConfig',
+  'descriptionConfig',
+  'commentsConfig',
+];
+
+export const isAnyRequiredFieldNotSet = (mapping: Record<string, unknown> | undefined) =>
+  !casesRequiredFields.some((field) => mapping != null && mapping[field] != null);
+
+/**
+ * The user can use either a connector of type cases or all.
+ * If the connector is of type all we should check if all
+ * required field have been configured.
+ */
+
+export const connectorValidator = (
+  connector: CaseActionConnector
+): ReturnType<ValidationConfig['validator']> => {
+  const {
+    config: { mappings, connectorType },
+  } = connector;
+  if (connectorType === SwimlaneConnectorType.Alerts || isAnyRequiredFieldNotSet(mappings)) {
+    return {
+      message: '',
```

Comment:
You are right about it 🙂 . The UX asked for the message to be empty because we already showing a callout. I have to return an empty message. Otherwise, the validator will not invalidate the form.

Reply:
The problem is that this validator is also used by our HTTP APIs, so if the connector goes through this validator, and goes through this code path, the HTTP response's error message won't indicate what's wrong.  It would be something like "invalid data when creating connector:" - where there should be the message from the validator at the end.

Could we change this so the UI "hides" this message somehow, if it's not desired to be seen?

Reply:
@pmuellr This validator is not used by actions. It is not part of the action's registration code. It is only used by cases in the case creation form. It is a `form` validator. Nevertheless, you are right! It is best to return something and hide it on the UI. I did it like that.